Sources/MapboxMaps/Foundation/MapView.swift

```diff
@@ -464,47 +421,33 @@ open class MapView: UIView {
         }
     }
 
-    private func updateDisplayLinkPreferredFramesPerSecond() {
-        if let displayLink = displayLink {
-            if let _preferredFramesPerSecond = _preferredFramesPerSecond {
-                displayLink.preferredFramesPerSecond = _preferredFramesPerSecond
-            }
-            // Checking Swift version as a proxy for iOS SDK version to enable
-            // building with iOS SDKs < 15
-            #if swift(>=5.5)
-            if #available(iOS 15.0, *) {
-                if let _preferredFrameRateRange = _preferredFrameRateRange {
-                    displayLink.preferredFrameRateRange = _preferredFrameRateRange
-                }
-            }
-            #endif
-        }
-    }
-
-    open override func willMove(toWindow newWindow: UIWindow?) {
-        super.willMove(toWindow: newWindow)
-        if newWindow != nil {
-            validateDisplayLink()
-        }
-    }
-
     open override func didMoveToWindow() {
         super.didMoveToWindow()
 
-        if window != nil {
-            validateDisplayLink()
-            subscribeToLifecycleNotifications()
-        } else {
-            unsubscribeFromLifecycleNotifications()
-            // TODO: Fix this up correctly.
-            displayLink?.invalidate()
-            displayLink = nil
+        unsubscribeFromLifecycleNotifications()
+        displayLink.invalidate()
+
+        let oldDisplayLink = displayLink
+        let newDisplayLink = dependencyProvider.makeDisplayLink(
+            window: window,
+            target: ForwardingDisplayLinkTarget { [weak self] in
+                self?.updateFromDisplayLink(displayLink: $0)
+            },
+            selector: #selector(ForwardingDisplayLinkTarget.update(with:)))
+        displayLink = newDisplayLink ?? EmptyDisplayLink()
+
+        // Checking Swift version as a proxy for iOS SDK version to enable
+        // building with iOS SDKs < 15
+#if swift(>=5.5)
+        if #available(iOS 15.0, *) {
+            displayLink.preferredFrameRateRange = oldDisplayLink.preferredFrameRateRange
         }
-    }
+#endif
+        displayLink.preferredFramesPerSecond = oldDisplayLink.preferredFramesPerSecond
```

Comment:
On line 443, we set `preferredFrameRateRange`, and here, we set `preferredFramesPerSecond`. Do they interact in any way? What happens if, on a real CADisplayLink, you set conflicting values? Does the order in which you set the conflicting values matter?

Reply:
It seems that we can't avoid checking if the preferred frame rate is overridden. The `EmptyDisplayLink` concept doesn't seem to work in this setup after all. Let's scrap it and move on with the original refactoring then.